Orange/widgets/utils/itemmodels.py

```diff
@@ -127,7 +127,10 @@ def mapToSourceRows(self, rows):
             Source rows matching input rows. If they are the same,
             simply input `rows` is returned.
         """
-        if self.__sortInd is not None:
+        # self.__sortInd[rows] fails if `rows` is an empty list or array
+        if self.__sortInd is not None \
+                and (isinstance(rows, (int, type(Ellipsis)))
+                     or len(rows)):
```

Comment:
@kernc, what do you prefer:

- `(isinstance(rows, (int, type(Ellipsis))) or len(rows))`,
- `(not isinstance(rows, (list, np.ndarray)) or len(rows))`, or
- `(not isinstance(rows, collections.Sized) or len(rows))`?

If the method is called with the argument types specified in the docstring, all are equivalent. Also, in cases where they are not equivalent (e.g. if `rows` is something stupid), the next line will fail anyway.

Reply:
I already see:
```py
>>> arr = np.ones(5)
>>> empty_list = []
>>> arr[empty_list]
array([], dtype=float64)
```
I.e. empty input, empty output.

You don't?

Reply:
I would prefer:
```py
if self.__sortInd is not None:
    if rows is not Ellipsis:
        rows = np.asarray(rows, dtype=int)

    new_rows = self.__sortInd[rows]
    ...
```
But this also relies on indexing (_at least_) with an empty _array_ working.

Reply:
_I_ would prefer your first option. Note to also fix `.mapFromSourceRows()` below.

Reply:
dtype must not be float; the caller must ensure that. This is similar to `__getitem__` which expects ints, and doesn't cast float to int. I made this clear in the docstring.

I fixed `mapFromSourceRows()`.